src/parser/meta.rs

```diff
@@ -254,15 +257,16 @@ pub fn take_until_size(buf: &[u8], byte_size: u32) -> IResult<&[u8], &[u8]> {
     Ok((remaining, extracted))
 }
 
-fn parse_meta_flag_values_as_slice(input: &[u8]) -> IResult<&[u8], Vec<(u8, &[u8])>> {
+#[allow(clippy::type_complexity)]
+fn parse_meta_flag_values_as_slice(input: &[u8]) -> IResult<&[u8], Vec<(u8, Option<&[u8]>)>> {
     if input.is_empty() || input == b"\r\n" {
         Ok((input, Vec::new()))
     } else {
         many0(map(
             tuple((
                 space1,
                 map(take(1usize), |s: &[u8]| s[0]),
-                take_while1(|c: u8| c != b'\r' && c != b' '), // finding a space means more flags, finding \r means end of flags
+                opt(take_while1(|c: u8| c != b'\r' && c != b' ')), // finding a space means more flags, finding \r means end of flags
```

Comment:
Needed to add this in when encountering single character tokens like X, W, Z which never have anything following them.  This was leading to nom errors in any case that they showed up.